Sources/Basic/TerminalController.swift

```diff
@@ -0,0 +1,252 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import libc
+import func POSIX.getenv
+
+private protocol OutputFileStreamProtocol: OutputStream {
+    var fd: Int32 { get }
+    var file: UnsafeMutablePointer<FILE> { get }
+}
+
+final public class OutputFileStream: OutputFileStreamProtocol {
+    private let fd: Int32
+    private let file: UnsafeMutablePointer<FILE>
+
+    init(file: UnsafeMutablePointer<FILE>) {
```

Comment:
FWIW, the goal is that OutputByteStream have support for writing directly to a file. We should probably do that rather than expose two similar methods with different APIs.

For that to happen, `OutputByteStream` will need to get specific classes which implement the in-memory stream versus the to-file stream, and we will need to move some of the accessors (the ones which return the full contents) to the in-memory stream. The to-file stream can expose methods to get at the underlying file data.

This is very similar to how LLVM's `raw_ostream` is structured, which is what that class is modeled after.